Makefile

```diff
@@ -12,3 +12,21 @@ proto-lint:
 	@echo "--> Linting Protobuf files"
 	@$(DOCKER_BUF) lint
 .PHONY: proto-lint
+
+## lint: Lint source code. Requires golangci-lint.
+lint:
+	@echo "--> Lint source code using golangci-lint"
+	@golangci-lint run
+.PHONY: lint
```

Comment:
[optional] we can add a section to the README that instructs developers to install golangci-lint. Something like https://github.com/celestiaorg/celestia-app/blob/b1592c314d4dca77c477e62ee1890970c6ee6e0c/README.md?plain=1#L119-L121

Reply:
@rootulp I have modified the same PR with readme as well as help command. Let me know your feedback. for the Makefile, i have followed the default format, and the commands would refer to .PHONY.

Reply:
thanks, we appreciate it!